graphql_compiler/compiler/expressions.py

```diff
@@ -794,3 +794,58 @@ def to_gremlin(self):
             predicate=self.predicate.to_gremlin(),
             if_true=self.if_true.to_gremlin(),
             if_false=self.if_false.to_gremlin())
+
+
+class BetweenClause(Expression):
+    """A `BETWEEN` Expression, constraining a field value to lie within a lower and upper bound."""
+
+    def __init__(self, field, lower_bound, upper_bound):
+        """Construct an expression that is true when the field value is within the given bounds.
+
+        Args:
+            field: LocalField Expression, denoting the field in consideration
+            lower_bound: lower bound constraint for given field
+            upper_bound: upper bound constraint for given field
+        Returns:
+            a new BetweenClause object
+        """
+        super(BetweenClause, self).__init__(self, field, lower_bound, upper_bound)
+        self.field = field
```

Comment:
use underscores for the private variables?

Reply:
I was following https://github.com/kensho-technologies/graphql-compiler/blob/optional_traversal/graphql_compiler/compiler/expressions.py#L613.
These are not necessarily private variables I think.

Reply:
Yeah this isn't necessarily private, lowering steps might refer to this variable.